whois-query/src/main/java/net/ripe/db/whois/query/acl/AccessControlListManager.java

```diff
@@ -55,40 +73,61 @@ public boolean requiresAcl(final RpslObject rpslObject, final Source source) {
                 || (ObjectType.ROLE.equals(objectType) && rpslObject.findAttributes(AttributeType.ABUSE_MAILBOX).isEmpty());
     }
 
-    public boolean isDenied(final InetAddress remoteAddress) {
-        return resourceConfiguration.isDenied(remoteAddress);
-    }
+    public boolean isDenied(final InetAddress remoteAddress, final String ssoToken) {
+        if(ipResourceConfiguration.isDenied(remoteAddress)) {
+            return true;
+        }
 
-    public boolean isAllowedToProxy(final InetAddress remoteAddress) {
-        return resourceConfiguration.isProxy(remoteAddress);
+        final String username = getUserName(ssoToken);
+        return username != null ? ssoResourceConfiguration.isDenied(username) : false;
     }
 
-    int getPersonalDataLimit(final InetAddress remoteAddress) {
-        return resourceConfiguration.getLimit(remoteAddress);
+    public boolean isAllowedToProxy(final InetAddress remoteAddress) {
+        return ipResourceConfiguration.isProxy(remoteAddress);
     }
 
     public boolean isUnlimited(final InetAddress remoteAddress) {
-        return getPersonalDataLimit(remoteAddress) < 0;
+        return ipResourceConfiguration.getLimit(remoteAddress) < 0;
     }
 
-    public boolean canQueryPersonalObjects(final InetAddress remoteAddress) {
-        return getPersonalObjects(remoteAddress) >= 0;
+    public boolean canQueryPersonalObjects(final InetAddress remoteAddress, final String ssoToken) {
+        return getPersonalObjects(remoteAddress,ssoToken) >= 0;
     }
 
     public boolean isTrusted(final InetAddress remoteAddress) {
         return ipRanges.isTrusted(IpInterval.asIpInterval(remoteAddress));
     }
 
-    public int getPersonalObjects(final InetAddress remoteAddress) {
+    public int getPersonalObjects(final InetAddress remoteAddress, final String ssoToken) {
```

Comment:
Overload `getPersonalObjects()` with only a `remoteAddress` parameter so you don't need to invoke it with `null` from other classes?

Reply:
Actually I wanted to have only one way to account for personal objects. So that we do not end up up calling a method for remoteAddr by mistake from somewhere even though sso is available. Plus lots of these services will eventually have sooToken available when db-webui changes are in place. Else we will end up having logic in all service calls that decide which method to call remoteAddr or sso